tests/datacache/test_datacache.py

```diff
@@ -34,3 +34,17 @@ def test_cache(tmp_path: "Path") -> None:
     assert len(cache) == 1
     cache.clear()
     assert len(cache) == 0
+
+
+def test_numpy(tmp_path: "Path") -> None:
+    """Test adding numpy arrays in the datachache."""
+    import numpy as np
+
+    from oteapi.datacache.datacache import DataCache
```

Comment:
```suggestion
    from oteapi.datacache import DataCache
```
You do not need the extra datacache in the import

Reply:
Thanks. Your suggestions are included.

In addition we do not check explicit for numpy, but tries to store directly in datacache by default. If it fails we fall back on json serialisation.

Reply:
Added an optional `json_encoder` argument to datacache.add(), with which one can supply a json encoder for the data one adds. 

Also added a test that shows how this can be used to add a ase.Atoms object to the datacache.